src/zope/app/apidoc/typemodule/tests.py

```diff
@@ -13,38 +13,47 @@
 ##############################################################################
 """Tests for the Book Documentation Module
 
-$Id$
 """
+
 import unittest
 import doctest
 
-from zope.app.testing.functional import BrowserTestCase
-from zope.app.testing import placelesssetup
+from zope.component import testing
+
+from zope.app.apidoc.tests import BrowserTestCase
+from zope.app.apidoc.tests import standard_checker
+from zope.app.apidoc.tests import standard_option_flags
 from zope.app.apidoc.testing import APIDocLayer
 
 
 class TypeModuleTests(BrowserTestCase):
     """Just a couple of tests ensuring that the templates render."""
 
+    layer = APIDocLayer
+
     def testMenu(self):
         response = self.publish(
             '/++apidoc++/Type/@@menu.html',
             basic='mgr:mgrpw')
         self.assertEqual(response.getStatus(), 200)
         body = response.getBody()
-        self.assert_(body.find('IBrowserSkinType') > 0)
+        self.assertIn('IBrowserSkinType', body)
         self.checkForBrokenLinks(body, '/++apidoc++/Type/@@menu.html',
                                  basic='mgr:mgrpw')
 
 
 def test_suite():
-    TypeModuleTests.layer = APIDocLayer
+    checker = standard_checker()
+
     return unittest.TestSuite((
-        doctest.DocTestSuite('zope.app.apidoc.typemodule.type',
-                             setUp=placelesssetup.setUp,
-                             tearDown=placelesssetup.tearDown),
-        unittest.makeSuite(TypeModuleTests),
-        ))
+        doctest.DocTestSuite(
+            'zope.app.apidoc.typemodule.type',
+            setUp=testing.setUp,
+            tearDown=testing.tearDown,
+            checker=checker,
+            optionflags=standard_option_flags),
+        unittest.defaultTestLoader.loadTestsFromName(__name__),
+    ))
 
 if __name__ == '__main__':
-    unittest.main(default='test_suite')
+    unittest.main()
```

Comment:
I wasn't paying attention to this in other test files, but -- why?

defaultTest='test_suite' is required so we won't be skipping half of our tests (all the doctests).

Then again it's not worth supporting running each unittest file individually via python; you can use zope.testrunner -m zope.app.apidoc.tests.test_foo to limit the test run to a single file.  Why not drop the entire `if __name__ == ...` check in all the test files?